pkg/virt-controller/watch/vm.go

```diff
@@ -1250,6 +1246,16 @@ func (c *VMController) cleanupRestartRequired(vm *virtv1.VirtualMachine) (*virtv
 }
 
 func (c *VMController) startVMI(vm *virtv1.VirtualMachine) (*virtv1.VirtualMachine, error) {
+	ready, err := c.handleDataVolumes(vm)
```

Comment:
What I meant in my previous comment was to add here before the handleDataVolumes:
if vm.Annotations[virtv1.ImmediateDataVolumeCreation] != "true" {
which make the handleDataVolumes be called only if it wasnt called already in the `sync` main function

Reply:
`handleDataVolumes` not exactly just looks in memory it also creates datavolumes. if you create the dvs and a second later call the function again, it might be possible that they will not exist in the cache yet(?) and call `create` again which will fail on dv already exists..
I dont think removing an `If` later on if you also deprecate the `ImmediateDataVolumeCreation` annotation makes maintenance/refactoring much more difficult :)

Reply:
Given `vm.Annotations[virtv1.ImmediateDataVolumeCreation] != "false"` (the usual case) and the current code path, when `handleDataVolumes` is called in `startVMI` this code will have already executed [1] meaning DVs are created and in the cache.  It will essentially be a no op.

Adding `if vm.Annotations[virtv1.ImmediateDataVolumeCreation] != "true"`  in `startVMI` will only optimize the no-op case.

Also the fewest spots that have to care about annotations the better for refactoring/maintenance.  

I just don't see much of a gain by making `startVMI` aware of `ImmediateDataVolumeCreation`

[1] https://github.com/mhenriks/kubevirt/blob/no-dvs-when-stopped/pkg/virt-controller/watch/vm.go#L3117-L3128

Reply:
OK I won't insist on that.